octodns/manager.py

```diff
@@ -275,7 +275,10 @@ def sync(self, eligible_zones=[], eligible_targets=[], dry_run=True,
             self.log.info('sync:   sources=%s -> targets=%s', sources, targets)
 
             try:
-                sources = [self.providers[source] for source in sources]
+                collected = []
+                for source in sources:
+                    collected.append(self.providers[source])
+                sources = collected
```

Comment:
Why?

Reply:
Don't remember off-hand. Don't think I would have changed it unless it was fixing a test failure.

Reply:
Figured out what was going on with this, I think. Looks like the `except` block was throwing an exception and causing the test for this bit to fail. It did capture an exception, but not the expected message. It'd probably make sense to throw a `ManagerException`, something specific to avoid this confusion/case so going to look at that change.